docs-starlight/src/content/docs/02-features/05-run-queue.mdx

```diff
@@ -24,57 +24,247 @@ Terragrunt analyzes these dependencies to determine the correct order of operati
+
+  e.g. `terragrunt run --all plan --queue-include-dir "subtree/*"`
+
+  Include units within the `subtree` directory (along with their dependencies), i.e., `subtree/dependent`, `subtree/dependency` and `ancestor-dependency`.
+
+  <Aside type="note">
+  Using the `--queue-include-dir` automatically triggers "Exclude by default" behavior, as mentioned above.
+  </Aside>
+
+  <Aside type="note">
+  `ancestor-dependency` is still included by default here because it's a dependency of `dependency`.
+
+  Using [`--queue-strict-include`](/docs/reference/cli/commands/run#queue-strict-include) would prevent that.
+  </Aside>
+
+- [`--queue-exclude-dir`](/docs/reference/cli/commands/run#queue-exclude-dir): Specify glob patterns for directories to *exclude*. Can be specified multiple times.
+
+  e.g. `terragrunt run --all plan --queue-exclude-dir "independent"`
+
+  Exclude the `independent` unit. `ancestor-dependency`, `subtree/dependency`, and `subtree/dependent` would still be processed according to their dependencies.
+
+  <Aside type="note">
+    Dependencies of excluded units will still be included unless they are also explicitly excluded.
+
+    In this example, excluding `subtree/dependency` would not automatically exclude `ancestor-dependency`.
+  </Aside>
+
+- [`--queue-excludes-file`](/docs/reference/cli/commands/run#queue-excludes-file): Provide a file containing a list of directories to exclude.
+
+  e.g. `terragrunt run --all plan --queue-excludes-file ".tg-excludes"`
+
+  ```text
+  # .tg-excludes
+
+  independent
+  subtree/dependency
+  ```
+
+  Exclude `independent` and `subtree/dependency` from the run, only running `subtree/dependent` and `ancestor-dependency`.
+
+  <Aside type="tip">
+  The default value for this flag is `.terragrunt-excludes`.
+
+  Populate a file named this in your project root to exclude units from being run by default, without using the `--queue-excludes-file` flag.
+  </Aside>
+
+- [`--queue-strict-include`](/docs/reference/cli/commands/run#queue-strict-include): Only include units matching `--queue-include-dir`.
+
+  e.g. `terragrunt run --all plan --queue-include-dir "subtree/dependency" --queue-strict-include`
+
+  Only include the `subtree/dependency` unit. Its dependency, `ancestor-dependency`, will be excluded because it does not match the include and strict mode is enabled.
+
+- [`--queue-include-external`](/docs/reference/cli/commands/run#queue-include-external): Include external dependencies (those outside the current working directory) by default.
+
+  e.g. `terragrunt run --all plan --working-dir subtree --queue-include-external`
+
+  Include `ancestor-dependency` in addition to the `subtree/dependent` and `subtree/dependency` units.
+
+  <Aside type="note">
+  The `--queue-include-external` flag is simply a convenience flag to avoid the interactive prompt to request inclusion of external dependencies.
```

Comment:
- What is the behaviour with terragrunt non-interactive?

Reply:
It's documented in the non-interactive docs, but I can document it here too:
https://terragrunt.gruntwork.io/docs/reference/cli-options/#non-interactive